web/src/data/recoil.ts

```diff
@@ -110,12 +110,38 @@ export const vaporConnectionConfig = selector<ConnectionConfig | undefined>({
   },
 });
 
+export const portalConnectionConfig = selector<ConnectionConfig | undefined>({
+  key: "portalConnectionConfig",
+  get: async ({ get }) => {
+    const {search} = window.location;
+    if (search) {
+      const queryParams = new URLSearchParams(search);
+      const portalHostname = queryParams.get("hostname");
+      const portalCredentials = queryParams.get("credentials");
+
+      if (portalCredentials) {
+        const decodedCredentials = atob(portalCredentials);
+        const { username, password } = JSON.parse(decodedCredentials);
```

Comment:
* what happens if decodedCredentials contains invalid JSON?
* what happens if decodedCredentials contains valid JSON, but not the expected object
* consider using `u` and `p` instead of `username` and `password` to reduce the length of the credentials blob

Reply:
added a try/catch to avoid fatal error

Reply:
I think it's more clear to use `username` and `password` and I don't think the blob length is a big issue, but let me know if you have a strong opinion here